mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/InformationToEmail.java

```diff
@@ -0,0 +1,200 @@
+            "This will result in a degraded service.\n" +
+            "To mitigate this issue you might reach your administrator in order to increase your configured quota. " +
+            "You might also delete some non important emails.";
+
+        private final StringBuilder stringBuilder;
+
+
+        public MessageBuilder() {
+            this.stringBuilder = new StringBuilder();
+        }
+
+        public MessageBuilder appendSizeReport(QuotaThreshold threshold, Quota<QuotaSize> sizeQuota) {
+            stringBuilder.append(String.format("You currently occupy more than %d %% of the total size allocated to you.\n" +
+                    "You currently occupy %s on a total of %s allocated to you.\n\n",
+                threshold.getQuotaOccupationRatioAsPercent(),
+                FileUtils.byteCountToDisplaySize(sizeQuota.getUsed().asLong()),
+                FileUtils.byteCountToDisplaySize(sizeQuota.getLimit().asLong())));
+            return this;
+        }
+
+        public MessageBuilder appendCountReport(QuotaThreshold threshold, Quota<QuotaCount> countQuota) {
+            stringBuilder.append(String.format("You currently occupy more than %d %% of the total message count allocated to you.\n" +
+                    "You currently have %d messages on a total of %d allowed for you.\n\n",
+                threshold.getQuotaOccupationRatioAsPercent(),
+                countQuota.getUsed().asLong(),
+                countQuota.getLimit().asLong()));
+            return this;
+        }
+
+        public MessageBuilder appendSizeReport(Optional<QuotaThreshold> threshold, Quota<QuotaSize> sizeQuota) {
+            if (threshold.isPresent()) {
+                return appendSizeReport(threshold.get(), sizeQuota);
+            }
+            return this;
+        }
+
+        public MessageBuilder appendCountReport(Optional<QuotaThreshold> threshold, Quota<QuotaCount> countQuota) {
+            if (threshold.isPresent()) {
+                return appendCountReport(threshold.get(), countQuota);
+            }
+            return this;
+        }
+
+        public MessageBuilder appendPreamble() {
+            stringBuilder.append(PREAMBLE);
+            return this;
+        }
+
+        public MessageBuilder appendConclusion() {
+            stringBuilder.append(CONCLUSION);
+            return this;
+        }
+
+        public String build() {
+            return stringBuilder.toString();
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    private final Optional<QuotaThreshold> countThreshold;
+    private final Optional<QuotaThreshold> sizeThreshold;
+    private final Quota<QuotaSize> sizeQuota;
+    private final Quota<QuotaCount> countQuota;
+
+    @VisibleForTesting
+    InformationToEmail(Optional<QuotaThreshold> countThreshold, Optional<QuotaThreshold> sizeThreshold,
```

Comment:
InformationToEmail is a strange name regarding that it refers only to Threasholds handling